src/aggregated_bls/party_i.rs

```diff
@@ -52,11 +57,41 @@ impl Keys {
 
     pub fn combine_local_signatures(sigs: &[SIG]) -> BLSSignature {
         let (head, tail) = sigs.split_at(1);
-        let sig_sum = tail.iter().fold(head[0], |x, acc| x + acc);
+        let sig_sum = tail.iter().fold(head[0], |acc, x| acc + x);
         BLSSignature { sigma: sig_sum }
     }
 
     pub fn verify(signature: &BLSSignature, message: &[u8], apk: &APK) -> bool {
         signature.verify(message, apk)
     }
+
+    fn efficient_pairing_loop(vec_g1: &[GE1], vec_g2: &[GE2]) -> Pair {
```

Comment:
use https://github.com/ZenGo-X/curv/blob/master/src/elliptic/curves/bls12_381/mod.rs#L30

Reply:
I changed it for compute_pairing for now. We could consider adding something similar for computing an efficient pairing of a list of pairs to the curv library instead:
```
fn efficient_pairing_loop(vec_g1: &[GE1], vec_g2: &[GE2]) -> Pair {
        let vec_g1_prep: Vec<_> = vec_g1.iter().map(|x| x.get_element().prepare()).collect();
        let vec_g2_prep: Vec<_> = vec_g2.iter().map(|x| x.get_element().prepare()).collect();
        let vec: Vec<_> = vec_g1_prep.iter().zip(vec_g2_prep.iter()).collect();
        Pair {
            e: Bls12::final_exponentiation(&Bls12::miller_loop(vec.iter())).unwrap(),
        }
} 
```